qiskit_optimization/translators/docplex.py

```diff
@@ -0,0 +1,288 @@
+                # should never happen
+                raise QiskitOptimizationError('Unsupported variable type: {}'.format(x.vartype))
+
+        # add objective
+        objective = quadratic_program.objective.constant
+        for i, v in quadratic_program.objective.linear.to_dict().items():
+            objective += v * var[cast(int, i)]
+        for (i, j), v in quadratic_program.objective.quadratic.to_dict().items():
+            objective += v * var[cast(int, i)] * var[cast(int, j)]
+        if quadratic_program.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            mdl.minimize(objective)
+        else:
+            mdl.maximize(objective)
+
+        # add linear constraints
+        for i, l_constraint in enumerate(quadratic_program.linear_constraints):
+            name = l_constraint.name
+            rhs = l_constraint.rhs
+            if rhs == 0 and l_constraint.linear.coefficients.nnz == 0:
+                continue
+            linear_expr = 0
+            for j, v in l_constraint.linear.to_dict().items():
+                linear_expr += v * var[cast(int, j)]
+            sense = l_constraint.sense
+            if sense == Constraint.Sense.EQ:
+                mdl.add_constraint(linear_expr == rhs, ctname=name)
+            elif sense == Constraint.Sense.GE:
+                mdl.add_constraint(linear_expr >= rhs, ctname=name)
+            elif sense == Constraint.Sense.LE:
+                mdl.add_constraint(linear_expr <= rhs, ctname=name)
+            else:
+                # should never happen
+                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+        # add quadratic constraints
+        for i, q_constraint in enumerate(quadratic_program.quadratic_constraints):
+            name = q_constraint.name
+            rhs = q_constraint.rhs
+            if rhs == 0 \
+                    and q_constraint.linear.coefficients.nnz == 0 \
+                    and q_constraint.quadratic.coefficients.nnz == 0:
+                continue
+            quadratic_expr = 0
+            for j, v in q_constraint.linear.to_dict().items():
+                quadratic_expr += v * var[cast(int, j)]
+            for (j, k), v in q_constraint.quadratic.to_dict().items():
+                quadratic_expr += v * var[cast(int, j)] * var[cast(int, k)]
+            sense = q_constraint.sense
+            if sense == Constraint.Sense.EQ:
+                mdl.add_constraint(quadratic_expr == rhs, ctname=name)
+            elif sense == Constraint.Sense.GE:
+                mdl.add_constraint(quadratic_expr >= rhs, ctname=name)
+            elif sense == Constraint.Sense.LE:
+                mdl.add_constraint(quadratic_expr <= rhs, ctname=name)
+            else:
+                # should never happen
+                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+        return mdl
+
+    def model_to_qp(self, model: Model, quadratic_program: Any):
```

Comment:
We could avoid having a `quadratic_program` as a method parameter if we made `from_model` static.
And should we have correct type hints?

Reply:
Yes, making `from_model` static is an option.

Reply:
should just return a `QuatraticProgram`, doesn't need to be static for that (and I'd keep it non-static, maybe there are cases where we would like to parametrize the translator). Same for GUROBI and interface.

Reply:
@stefan-woerner I mean `QuadraticProgram.from_model()` to be static, not the methods in the translators. 
I think it should look like: `qp = QuadraticProgram.from_model(another_model, translator)`

Reply:
@adekusar-drl thanks, sounds good!